### PR TITLE
docs: DLT-2002 fix remaining modal scrollbar issues

### PR DESCRIPTION
**Pull request by braddialpad**
# docs: DLT-2002 fix remaining modal scrollbar issues

## Obligatory GIF (super important!)

Obligatory GIF

## :hammer_and_wrench: Type Of Change

These types will not increment the version number, but will still deploy to documentation site on release:

- [x] Documentation

## :book: Jira Ticket

https://dialpad.atlassian.net/browse/DLT-2002

## :book: Description

- Fixed modal example component to use vue component instead of pure HTML/CSS
- Fixed modal component not removing the scrollbar on open
- Fixed algolia search not removing the scrollbar on open
- Moved rootScrolling functions to common so any package can use them.
- Added functionality to rootScrolling functions to support shadowDom

## :bulb: Context

Modal components were no longer correctly hiding the scrollbar since we implemented the dt-scrollbar

## :pencil: Checklist

For all PRs:

- [x] I have ensured no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).
- [x] I have reviewed my changes.
- [x] I have added all relevant documentation.
- [x] I have considered the performance impact of my change.

For all Vue changes:

- [x] I have made my changes in Vue 2 and Vue 3. Note: you may sync your changes from Vue 2 to Vue 3 (or vice versa) using the `./scripts/dialtone-vue-sync.sh` script. Read docs here: [Dialtone Vue Sync Script](../packages/dialtone-vue3/.github/CONTRIBUTING.md#dialtone-vue-sync-script)

## :link: Sources

https://kingsora.github.io/OverlayScrollbars/

### CODE REVIEW

File: apps/dialtone-documentation/docs/.vuepress/exampleComponents/ExampleModal.vue [hunk 1]
```diff
@@ -1,224 +1,84 @@
...
-    @keydown.esc="closeModal"
+  <dt-modal
+    title="Example title"
+    :close-button-props="{ ariaLabel: 'Close modal' }"
+    :show="isOpen"
+    :banner-title="bannerTitle"
+    :banner-kind="bannerKind"
+    :fixed-header-footer="fixedHeaderFooter"
+    :size="size"
+    :copy="modalDescription"
```
Comment (ninamarina) [inline apps/dialtone-documentation/docs/.vuepress/exampleComponents/ExampleModal.vue]: Previously, in the case where "fixedHeaderFooter" prop was true, we had the copy repeated to show that the header and footer are fixed and only the content is scrollable. I think we should keep that.

Comment (braddialpad) [reply apps/dialtone-documentation/docs/.vuepress/exampleComponents/ExampleModal.vue]: ah yeah good catch, I'll add that back.

**Comment by ninamarina** [comment]
I can still scroll while the algolia search modal is open:

algolia-scroll

File: apps/dialtone-documentation/docs/.vuepress/exampleComponents/ExampleModal.vue (hunk 1 above)
Comment (braddialpad) [reply apps/dialtone-documentation/docs/.vuepress/exampleComponents/ExampleModal.vue]: This one is fixed

**Comment by braddialpad** [comment]
> I can still scroll while the algolia search modal is open:

Well... This was actually pretty hard to make work. I could trigger the scrollbar disable when clicking the search button easily, but because the dialog is completely controlled by algolia and they don't appear to trigger any capturable close event it was quite difficult to enable the scrollbar again when closed.

I ended up using what I think is a pretty clever hack using focusout events on that dialog window to determine when it is closed (relatedEvent === null). Have a look, let me know what you think.

**Comment by ninamarina** [comment]
> > I can still scroll while the algolia search modal is open:
> 
> Well... This was actually pretty hard to make work. I could trigger the scrollbar disable when clicking the search button easily, but because the dialog is completely controlled by algolia and they don't appear to trigger any capturable close event it was quite difficult to enable the scrollbar again when closed.
> 
> I ended up using what I think is a pretty clever hack using focusout events on that dialog window to determine when it is closed (relatedEvent === null). Have a look, let me know what you think.

It works much better now, the only detail is that when changing tabs or focusing another app and going back to the site the scroll would now be enabled, I don't think it's serious anyway since it's an edge case.
The other problem that I found is that the sidebar is cropped so it's not visible if the search is open in the bottom of the page:

image

**Comment by braddialpad** [comment]
> he only detail is that when changing tabs or focusing another app and going back to the site the scroll would now be enabled, I don't think it's serious anyway since it's an edge case.
> The other problem that I found is that the sidebar is cropped so it's not visible if the search is open in the bottom of the page:

Hmm. yeah this is a bit janky I'll see if I can improve it.

**Comment by braddialpad** [comment]
Alright, changed it to use a mutation observer instead. Should work much more reliably now.

**Review by ninamarina — APPROVED** [review]
Nice, works as expected now

**Pull request merged**